velox/row/UnsafeRowDeserializers.h

```diff
@@ -855,6 +855,18 @@ struct UnsafeRowDeserializer {
       memory::MemoryPool* pool) {
     return convertToVectors(getBatchIteratorPtr(data, type), pool);
   }
+
+  /// Only supports deserialize Row with primitive type now.
+  /// @param memoryAddress the start memory address of the serailized UnsafeRow.
+  /// @param type the element type.
+  /// @param offsets offset of the row deserialized data. It's size should be
+  /// equal to the deserialized rows. First offset is 0.
+  /// @param pool the memory pool to allocate Vectors.
+  static VectorPtr deserialize(
```

Comment:
Is it necessary to keep two APIs for the deserialization instead of replacing the original one?

Reply:
Yes. This API only supports scalar types, it is hard to support complex data type, because it is hard to get the `offsets` for them. Another exists API supports all the data types.

Reply:
Can the APIs be unified so that only complex types use the row-row deserialization?

Reply:
I will try to implement complex data type, if success, we can compare the performance and decide if can discard the original implementation.

Reply:
yes, ok, I will do the refactor.

Reply:
The previous API read the rows from several `std::string_view`, but the new API requires all the row in one buffer started with `memoryAddress`, for this public override API, it supplies `ByteInputStream` which consists several buffers, so we cannot get all the rows in one buffer, so I keep the two APIs but the new API supports the complex data type by revoking the old API.
https://github.com/facebookincubator/velox/blob/main/velox/serializers/UnsafeRowSerializer.cpp#L135

Reply:
Users could use any of them as they need.

Reply:
And here requires the old API, because the serialize format is
| row size | serialized row | row size | serialized row
https://github.com/facebookincubator/velox/blob/main/velox/serializers/UnsafeRowSerializer.cpp#L145

This optimization requires the serialized data is in contiguous memory.

So we should keep the 2 APIs.